### PR TITLE
Closes #1639 create vignette for pk programming

### LINKED ISSUE #1639: Documentation:  Create vignette for PK ADNCA/ADPC programming
### Please select a category the issue is focused on?

User Guides

### Let us know where something needs a refresh or put your idea here!

Create new vignette for PK programming for ADNCA/ADPC.  This will correspond to the new template program.

### PR DESCRIPTION
**Pull request by jeffreyad**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by jeffreyad** [comment]
This new vignette for PK ADNCA/ADPC programming requires the update to add nominal time to `create_single_dose_dataset()` #1643 #1640

**Comment by bms63** [comment]
Hey @jeffreyad lots of merge conflicts to fix.  Let me know if you need help.

File: vignettes/pk_adnca.Rmd [hunk 1]
```diff
@@ -0,0 +1,928 @@
...
+dataset_vignette(
+  adpc,
+  display_vars = exprs(
+    USUBJID, PCTEST, ADTM, VISIT, PCTPT, NFRLT
+  )
+)
+```
+
+Next we will also join `ADSL` data with `EX` and derive dates/times.
+This section uses the `{admiral}` functions `derive_vars_merged`,
```
Comment (rossfarrugia) [inline vignettes/pk_adnca.Rmd:195]: include `()` after each function name - won't repeat this comment again, trust you to double check and update any cases

**Review by rossfarrugia — CHANGES_REQUESTED** [review]
@jeffreyad nice vignette!! I'm not a PK expert so i only added some minor general updates for conventions we've tended to follow in the other vignettes/templates. Apologies for late comments, but some like removing the default argument lines you probably want to match in the template too which i saw was already merged at https://github.com/pharmaverse/admiral/pull/1700, so maybe just include those updates in this branch/PR if you agree.

@bundfussr if you get chance probably worth you taking a look at this new vignette, as you might spot some useful enhancements too from your experience writing these for some of the other ADaMs.

File: vignettes/pk_adnca.Rmd [hunk 2]
```diff
@@ -0,0 +1,928 @@
...
+    by_vars = exprs(STUDYID, USUBJID)
+  ) %>%
+  # Derive analysis date/time
+  # Impute missing time to 00:00:00
+  derive_vars_dtm(
+    new_vars_prefix = "A",
+    dtc = PCDTC,
+    date_imputation = "FIRST",
+    time_imputation = "00:00:00",
+    ignore_seconds_flag = FALSE,
```
[the remark this answers is not in the document]
Comment (rossfarrugia) [reply vignettes/pk_adnca.Rmd:169]: @jeffreyad i still see these 2 arguments being set to same as default for these functions here and below - these lines could be removed

**Comment by jeffreyad** [comment]
> thanks @jeffreyad for the updates! will you make the same updates to the adpc template to match here? good practice to keep our ADaM vignettes and respective template consistent.

Sure, thanks @rossfarrugia

**Comment by bms63** [comment]
> > thanks @jeffreyad for the updates! will you make the same updates to the adpc template to match here? good practice to keep our ADaM vignettes and respective template consistent.
> 
> Sure, thanks @rossfarrugia

Does that mean you will drop a lot of the timing variables in that template?  I thought those were needed for the analysis?  

Thanks again for pushing these through @jeffreyad!!

**Comment by jeffreyad** [comment]
> > > thanks @jeffreyad for the updates! will you make the same updates to the adpc template to match here? good practice to keep our ADaM vignettes and respective template consistent.
> > 
> > 
> > Sure, thanks @rossfarrugia
> 
> Does that mean you will drop a lot of the timing variables in that template? I thought those were needed for the analysis?
> 
> Thanks again for pushing these through @jeffreyad!!

@bms63 No, it should just be removing the defaults for `flag_imputation`, `date_imputation`, and `ignore_seconds_flag`

**Comment by bms63** [comment]
@jeffreyad when you get a chance can you fix the man page errors.  I can re-approve so the templates fire off and then we can get this merged!!

**Review by rossfarrugia — APPROVED** [review]
Thanks for the updates @jeffreyad - fine from my perspective now, but will leave @bms63 to double check before merging.

**Comment by bms63** [comment]
Thanks @jeffreyad.  This new and updated templates are great addition for admiral!!

**Comment by thomas-neitmann** [comment]
Thank you so much for all the effort you put into creating this @jeffreyad 🥳 👏

**Pull request merged**